lib/prefab/client.rb

```diff
@@ -34,18 +26,23 @@ def initialize(options = Prefab::Options.new)
         @prefab_api_url = @options.prefab_api_url
         log_internal ::Logger::INFO, "Prefab Connecting to: #{@prefab_api_url}"
       end
+
+      context.clear
       # start config client
       config_client
     end
 
-    def with_log_context(lookup_key, properties)
-      Thread.current[:prefab_log_lookup_key] = lookup_key
-      Thread.current[:prefab_log_properties] = properties
+    def with_log_context(_lookup_key, properties, &block)
+      warn '[DEPRECATION] `$prefab.with_log_context` is deprecated.  Please use `with_context` instead.'
+      with_context(properties, &block)
+    end
+
+    def with_context(properties, &block)
```

Comment:
Name this `with_context(context, &block)` ?

If they accidentally pass in `with_context({name: "bob", team: "6"})` what happens? Just `properties` feels a bit like it's ok to free-for-all.

I think I'm onboard with this being PlainOldRuby and not a `Prefab::Context.create("user", 123, {subscription_level: "pro"})`  but I am still on the fence here. It feels like a way to give more guardrails.

Reply:
If you pass that in, you'll get an error from the `Context` initializer :) It tries to give a good error message about the format. Hopefully that + docs = 👍